cmd/integration/commands/idx_optimize.go

```diff
@@ -0,0 +1,199 @@
+package commands
+
+import (
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/erigontech/erigon-lib/common/background"
+	"github.com/erigontech/erigon-lib/config3"
+	lllog "github.com/erigontech/erigon-lib/log/v3"
+	"github.com/erigontech/erigon-lib/recsplit"
+	"github.com/erigontech/erigon-lib/recsplit/eliasfano32"
+	"github.com/erigontech/erigon-lib/recsplit/multiencseq"
+	"github.com/erigontech/erigon-lib/state"
+
+	"github.com/erigontech/erigon-lib/common"
+	"github.com/erigontech/erigon-lib/seg"
+	"github.com/erigontech/erigon/turbo/debug"
+	"github.com/spf13/cobra"
+)
+
+// TODO: this utility can be safely deleted after PR https://github.com/erigontech/erigon/pull/12907/ is rolled out in production
+func parseEFFilename(fileName string) (*efFileInfo, error) {
+	parts := strings.Split(fileName, ".")
+	stepParts := strings.Split(parts[1], "-")
+	startStep, err := strconv.ParseUint(stepParts[0], 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	endStep, err := strconv.ParseUint(stepParts[1], 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	return &efFileInfo{
+		prefix:    parts[0],
+		stepSize:  endStep - startStep,
+		startStep: startStep,
+		endStep:   endStep,
+	}, nil
+}
+
+type efFileInfo struct {
+	prefix    string
+	stepSize  uint64
+	startStep uint64
+	endStep   uint64
+}
+
+var b []byte
+
+func doConvert(baseTxNum uint64, v []byte) ([]byte, error) {
+	ef, _ := eliasfano32.ReadEliasFano(v)
+
+	seqBuilder := multiencseq.NewBuilder(baseTxNum, ef.Count(), ef.Max())
+	for it := ef.Iterator(); it.HasNext(); {
+		n, err := it.Next()
+		if err != nil {
+			return nil, err
+		}
+		seqBuilder.AddOffset(n)
+	}
+	seqBuilder.Build()
+
+	b = seqBuilder.AppendBytes(b[:0])
+	return b, nil
+}
+
+var idxOptimize = &cobra.Command{
+	Use:   "idx_optimize",
+	Short: "Scan .ef files, backup them up, reencode and optimize them, rebuild .efi files",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, _ := common.RootContext()
+		logger := debug.SetupCobra(cmd, "integration")
+
+		// accessorDir := filepath.Join(datadirCli, "snapshots", "accessor")
+		idxPath := filepath.Join(datadirCli, "snapshots", "idx")
```

Comment:
FYI: we have `dirs := datadir.New(datadirCli)`
and `dirs` object will have all paths. 
For example: `ilepath.Join(datadirCli, "snapshots", "idx")` can replace by `dirs.SnapIdx`